resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/PemsToKeyStoreConverter.scala

```diff
@@ -0,0 +1,117 @@
+   * to where the keyStore was written to disk.
+   */
+  def convertPemsToTempKeyStoreFile(
+      keyPemFile: File,
+      certPemFile: File,
+      keyAlias: String,
+      keyStorePassword: Option[String],
+      keyPassword: Option[String],
+      keyStoreType: Option[String]): File = {
+    require(keyPemFile.isFile, s"Key pem file provided at ${keyPemFile.getAbsolutePath}" +
+      " does not exist or is not a file.")
+    require(certPemFile.isFile, s"Cert pem file provided at ${certPemFile.getAbsolutePath}" +
+      " does not exist or is not a file.")
+    val privateKey = parsePrivateKeyFromPemFile(keyPemFile)
+    val certificate = parseCertificateFromPemFile(certPemFile)
+    val resolvedKeyStoreType = keyStoreType.getOrElse(KeyStore.getDefaultType)
+    val keyStore = KeyStore.getInstance(resolvedKeyStoreType)
+    keyStore.load(null, null)
+    keyStore.setKeyEntry(
+      keyAlias,
+      privateKey,
+      keyPassword.map(_.toCharArray).orNull,
+      Array(certificate))
+    val keyStoreOutputPath = Paths.get(s"keystore-${UUID.randomUUID()}.$resolvedKeyStoreType")
+    Utils.tryWithResource(new FileOutputStream(keyStoreOutputPath.toFile)) { storeStream =>
+      keyStore.store(storeStream, keyStorePassword.map(_.toCharArray).orNull)
+    }
+    keyStoreOutputPath.toFile
+  }
+
+  def convertCertPemToTrustStore(
+      certPemFile: File,
+      certAlias: String,
+      trustStoreType: Option[String]): KeyStore = {
+    require(certPemFile.isFile, s"Cert pem file provided at ${certPemFile.getAbsolutePath}" +
+      " does not exist or is not a file.")
+    val trustStore = KeyStore.getInstance(trustStoreType.getOrElse(KeyStore.getDefaultType))
+    trustStore.load(null, null)
+    val certificate = parseCertificateFromPemFile(certPemFile)
+    trustStore.setCertificateEntry(certAlias, certificate)
+    trustStore
+  }
+
+  private def withPemParsedFromFile[T](pemFile: File)(f: (PEMParser => T)): T = {
+    Utils.tryWithResource(new FileInputStream(pemFile)) { pemStream =>
+      Utils.tryWithResource(new InputStreamReader(pemStream, Charsets.UTF_8)) { pemReader =>
+        Utils.tryWithResource(new PEMParser(pemReader))(f)
+      }
+    }
+  }
+
+  private def parsePrivateKeyFromPemFile(keyPemFile: File): PrivateKey = {
+    withPemParsedFromFile(keyPemFile) { keyPemParser =>
+      val converter = new JcaPEMKeyConverter
+      keyPemParser.readObject() match {
+        case privateKey: PrivateKeyInfo =>
+          converter.getPrivateKey(privateKey)
+        case keyPair: PEMKeyPair =>
+          converter.getPrivateKey(keyPair.getPrivateKeyInfo)
+        case _ =>
+          throw new SparkException(s"Key file provided at ${keyPemFile.getAbsolutePath}" +
+            s" is not a key pair or private key PEM file.")
+      }
+    }
+  }
+
+  private def parseCertificateFromPemFile(certPemFile: File): X509Certificate = {
+    withPemParsedFromFile(certPemFile) { certPemParser =>
+      certPemParser.readObject() match {
```

Comment:
Although once again I think meaningfully testing the processing of multiple certificates is going to be hard. That once again is best left to the unit level.